flexx/app/_tornadoserver.py

```diff
@@ -678,13 +678,18 @@ def check_origin(self, origin):
         serving_host = self.request.headers.get("Host")
         serving_hostname = serving_host.split(':')[0]
         connecting_host = urlparse(origin).netloc
+        connecting_hostname = connecting_host.split(':')[0]
         
         if serving_hostname == 'localhost':
             return True  # Safe
         elif serving_hostname == '0.0.0.0':
             return True  # we cannot know if the origin matches
         elif serving_host == connecting_host:
             return True
+        elif serving_hostname == connecting_hostname:
+            # Comparing hostnames in case of Jupyter notebooks
+            # connecting_host and serving_host have same host names but different ports
+            return True
```

Comment:
ah ... that wont work, of course. The browser is connected to the Jupyter server via `host:8888`. The server and kernel maintain a ZMQ connection. Flexx needs its own connection (between browser and kernel), and thus its own port. 

Did you use `config.hostname` to make Flexx use the same hostname as the Jupyter server? I guess that would be the way to do it, since AFAIK Flexx does not have a way (from inside the kernel) to know what hostname the server is using.